### DISCUSSION TITLE
Welcome to cardano-docker Discussions!

### DISCUSSION THREAD

**Discussion by pascallapointe** [PaskLab/cardano-docker#12] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by d0nch4n** [comment]
Hi Pascal,

I'm new in the Raspberry PI and Cardano world.

I don't understand which are the steps to follow from scratch if I have an Ubuntu Server 64bit installed on my Pi4 8GB.

I have to install first Docker, but how can I use what you created by docker?

Thx a lot for your job.

**Reply by pascallapointe** [reply]
Hi, you're very welcome.

first make sure you also have docker-compose installed. It is not mandatory, but if you want take part of full automation, this is the way to go.

In short, my repository is a set of dockerfiles. Dockerfiles are a blue print to build an image. If you open them, you'll see that they mostly contain a sequence of linux instruction. You could basically compile the cardano node by following instructions in there (as long as you remove the RUN command in front of them, since it's a dockerfile command).

To use my setup, you just need to follow the steps described in the README file. Don't forget the subsection containing the monitoring docker image.

In the end you end up with a docker image containing the base setup for running a relay node or a pool. That image will be used to create one or more container. Each container could be a running relay on a same or different host.

The first step is to build the environment, that image is called cardano_env. The dockerfile for building it first download the ubuntu 20.04 image an than, build the GHC Haskell compilator for ARM on it and install all dependencies required for compiling the cardano-node binaries from source.

Then the next step is to build the cardano_node docker image using the Dockerfile for it. That dockerfile is using the cardano_env image created in step 1. They are separated in to parts because you will have to do step 2 everytimes a new cardano-node version is released. That allow you to do only the later part without the need of building the GHC compiler everytimes.

Once all that is done, you'll have a cardano-node docker image from which you can spin containers.
My instructions needs an update (there is 2 missing directories in my instruction), but in short you need to create a container from a directory container a `config`, `db` and `socket` folders. You will place all configurations for the node (topology, genesis files,key files, etc) in the config folder. The container "bind" to your local folder, so you can edit configuration without getting inside the container to change things.

If using docker-compose, which I strongly suggest, you'll be able to bring up a relay or bloc producer by using only one command: `docker-compose up -d`

Keep in mind that this provide only a way to keep your host system clean from build dependencies, manage versions and automate some part of a stake pool operations. But it doesn't do more, meaning that you still need to understand how to operate a cardano-node by hand.

If you are already proficient with cardano-node and cardano-cli, docker can bring you some value. If not, it might make it really harder to get your head around this whole things. You'll end-up trying to learn to 2 things that have a steep learning curve at the same time, which isn't a good idea. You should try successfully running a relay first before taking advantage of Docker, because docker will really get in your way when trying things out. (I don't know your current knowledge set, i'm just giving my 2 cents advice)

Hopes it answers some of your questions, docker isn't quite easy to explain with short sentences ..!

Cheers,

Pascal
